packages/http/src/validator/validators/params.ts

```diff
@@ -15,45 +16,42 @@ export class HttpParamsValidator<Target, Spec extends IHttpParam> implements IHt
 
   public validate(target: Target, specs: Spec[]): IPrismDiagnostic[] {
     const { _registry: registry, _prefix: prefix, _style: style } = this;
-    return specs.reduce<IPrismDiagnostic[]>((results, spec) => {
-      if (!(spec.name in target) && spec.required === true) {
-        results.push({
-          path: [prefix, spec.name],
-          code: 'required',
-          message: `Missing ${spec.name} ${prefix} param`,
-          severity: DiagnosticSeverity.Error,
-        });
-
-        // stop further checks
-        return results;
-      }
-
-      const resolvedStyle = spec.style || style;
-      if (spec.content && spec.content.schema && target[spec.name]) {
-        const deserializer = registry.get(resolvedStyle);
 
-        if (deserializer) {
-          Array.prototype.push.apply(
-            results,
-            validateAgainstSchema(
-              deserializer.deserialize(spec.name, target, spec.content.schema, spec.explode || false),
-              spec.content.schema,
-              prefix,
-            ),
+    const deprecatedWarnings = specs.filter(spec => spec.deprecated).map(spec => ({
+      path: [prefix, spec.name],
+      code: 'deprecated',
+      message: `${upperFirst(prefix)} param ${spec.name} is deprecated`,
+      severity: DiagnosticSeverity.Warning,
+    }));
+
+    const schema = createJsonSchemaFromParams(specs);
+
+    const parameterValues = pickBy(
+      mapValues(keyBy(specs, s => s.name.toLowerCase()), el => {
+        const resolvedStyle = el.style || style;
+        const deserializer = registry.get(resolvedStyle);
+        if (deserializer)
+          return deserializer.deserialize(
+            el.name.toLowerCase(),
+            target,
+            schema.properties && (schema.properties[el.name] as JSONSchema4),
+            el.explode || false,
           );
-        }
-      }
-
-      if (spec.deprecated === true) {
-        results.push({
-          path: [prefix, spec.name],
-          code: 'deprecated',
-          message: `${upperFirst(prefix)} param ${spec.name} is deprecated`,
-          severity: DiagnosticSeverity.Warning,
-        });
-      }
-
-      return results;
-    }, []);
+
+        return undefined;
+      }),
+    );
+
+    return validateAgainstSchema(parameterValues, schema, prefix).concat(deprecatedWarnings);
   }
 }
+
+function createJsonSchemaFromParams(params: IHttpParam[]): JSONSchema4 | JSONSchema6 | JSONSchema7 {
+  const schema: JSONSchema4 | JSONSchema6 | JSONSchema7 = {
+    type: 'object',
+    properties: pickBy(mapValues(keyBy(params, p => p.name.toLowerCase()), 'schema')) as JSONSchema4,
```

Comment:
Do you need to use `pickBy` here? What's the benefit? IIRC it would work the same way w/o it.

Reply:
`pickBy` will automatically remove any property that's undefined

Reply:
Oh cool stuff, good to know!